test/supplemental_genbank_tests/genbank_upload_full_test.py

```diff
@@ -152,18 +192,340 @@ def test_simple_shock_upload(self):
+        with pytest.raises(Exception) as got:
+            self.serviceImpl.genbanks_to_genomes(self.ctx, params)
+        assert_exception_correct(got.value, ValueError(error_message))
+
+    def _load_expected_data(self, json_path):
+        with open(json_path, "r") as read_file:
+            data = json.load(read_file)
+        return data
+
+    def _retrieve_provenance(self, provenance):
+        # make a copy to avoid modifying the original provenance
+        provs = [prov.copy() for prov in provenance]
+        for prov in provs:
+            for key in ["time", "epoch"]:
+                prov.pop(key)
+            for subaction in prov['subactions']:
+                subaction.pop("commit")
+        return provs
+
+    def _retrieve_genome_metadata(self, metadata):
+        # make a copy to avoid modifying the original metadata
+        metadata = dict(metadata)
+        metadata.pop("Assembly Object")
+        return metadata
+
+    def _retrieve_genome_data(self, data):
+        # make a copy to avoid modifying the original genome data
+        data = dict(data)
+        for key in ["assembly_ref", "genbank_handle_ref"]:
+            data.pop(key)
+        for key in ["cdss", "features", "mrnas", "non_coding_features"]:
+            for dist in data.get(key):
+                dist.pop("aliases", None)
+        for ontology_event in data.get("ontology_events", []):
+            ontology_event.pop("timestamp")
+            ontology_event.pop("ontology_ref")
+        return data
+
+    def _retrieve_assembly_data(self, data):
+        # make a copy to avoid modifying the original assembly data
+        data = dict(data)
+        data.pop("fasta_handle_ref")
+        fasta_handle_info = data["fasta_handle_info"]
+        fasta_handle_info.pop("shock_id")
+        for key in ["hid", "id"]:
+            fasta_handle_info["handle"].pop(key)
+        return data
+
+    def _get_object(self, result, is_genome):
+        ref = 'genome_ref' if is_genome else 'assembly_ref'
+        assert _OBJECT_VERSION_PATTERN.match(result[ref])
+        return self.wsClient.get_objects2({"objects": [{'ref': result[ref]}]})["data"][0]
+
+    def _check_info(self, obj, result, file_names, idx, expected_metadata, is_genome):
+        info = obj["info"]
+        object_info = 'genome_info' if is_genome else 'assembly_info'
+        object_name = file_names[idx] if is_genome else file_names[idx] + "_assembly"
+        object_type = 'KBaseGenomes.Genome' if is_genome else 'KBaseGenomeAnnotations.Assembly'
+        retrieved_metadata = self._retrieve_genome_metadata(info[10]) if is_genome else info[10]
+
+        assert info == result[object_info]
+        assert info[1] == object_name
+        assert info[2].split('-')[0] == object_type
+
+        # datetime.fromisoformat is not available in Python 3.6 or below
+        assert datetime.strptime(info[3], '%Y-%m-%dT%H:%M:%S+%f')
+        assert info[6] == self.wsID
```

Comment:
No version check?

Reply:
You mean upa check against genome ref?

Reply:
No, I just mean something like

```
assert info[4] == 1
```

Reply:
we have it here:https://github.com/kbaseapps/GenomeFileUtil/blob/0aaf9313236df8802f0d97cba267ac17eee10ac8/test/supplemental_genbank_tests/genbank_upload_full_test.py#L266

Or you think extra check is needed.

Reply:
That checks the upa, not the info

Reply:
👍

Reply:
:+1: